pandas/tests/io/json/test_ujson.py

```diff
@@ -780,8 +780,10 @@ def test_array_float(self):
         tm.assert_almost_equal(arr, arr_out)
 
     def test_0d_array(self):
-        with pytest.raises(TypeError):
+        msg = "array(1) (0d array) is not JSON serializable at the moment"
+        with pytest.raises(TypeError) as excinfo:
             ujson.encode(np.array(1))
+        assert str(excinfo.value) == msg
```

Comment:
Instead of doing this should be able to use the `match` argument of `pytest.raises`

Reply:
I tried that first, actually, but the need to escape the regex makes the expected output harder to readamd the error report is less informative.

Reply:
could use re.escape as we use this pattern elsewhere

```diff
diff --git a/pandas/tests/io/json/test_ujson.py b/pandas/tests/io/json/test_ujson.py
index 2cecf8f7b..d6572ac7b 100644
--- a/pandas/tests/io/json/test_ujson.py
+++ b/pandas/tests/io/json/test_ujson.py
@@ -780,10 +780,10 @@ class TestNumpyJSONTests:
         tm.assert_almost_equal(arr, arr_out)
 
     def test_0d_array(self):
-        msg = "array(1) (0d array) is not JSON serializable at the moment"
-        with pytest.raises(TypeError) as excinfo:
+        # gh-18878
+        msg = re.escape("array(1) (0d array) is not JSON serializable at the moment")
+        with pytest.raises(TypeError, match=msg):
             ujson.encode(np.array(1))
-        assert str(excinfo.value) == msg
 
     @pytest.mark.parametrize(
         "bad_input,exc_type,kwargs",

```